### PR TITLE
[Fix] Duplicate kill listener

### PR DESCRIPTION
**Pull request by BrettCleary**
Removes the duplicate 'kill' handler added in from download manager PR which causes issues with other ipcMain handlers.

Also removed the pre-push test command and added os checks to certain tests that are only written for linux.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/logger/__tests__/logfile.test.ts [hunk 1]
```diff
@@ -13,8 +14,9 @@ jest.mock('../logger')
 jest.unmock('../logfile')
 
 let tmpDir = {} as DirResult
+const skipIfOnWin = platform() === 'win32' ? describe.skip : describe
 
-describe('logger/logfile.ts', () => {
+skipIfOnWin('logger/logfile.ts', () => {
```
Comment (arielj) [inline src/backend/logger/__tests__/logfile.test.ts:19]: I wonder if it's better to do:

```
describe(..... () => {
  if (platform() === 'win32') {
    console.log('ignored on windows')
    return
  }

  ....
}
```

maybe it's just me though

**Review by arielj — APPROVED** [review]
Looks good to me, just added a comment but it's all good

**Comment by CommandMC** [comment]
The 2nd `kill` method is missing the extra functionality added by #1864. To resolve this, replace
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/1e01cb25693dc36ebcd6f346aaa05d28aa3dc24c/src/backend/main.ts#L1329
with
```ts
runner === 'sideload' ? stop(appName) : getGame(appName, runner).stop()
```
(import `stop` from `./sideload/games`)

**Review by Nocccer — APPROVED** [review]

**Review by Nocccer — DISMISSED** [review]
Aggree with @CommandMC. There is something missing from the sideload app. Easy fix.

File: src/backend/logger/__tests__/logfile.test.ts (hunk 1 above)
Comment (BrettCleary) [reply src/backend/logger/__tests__/logfile.test.ts:19]: yeah this will make it easier to find the tests that are being skipped. I'll add this

**Review by flavioislima — APPROVED** [review]
Looks good. Tested here and now Login is working again.
Thanks for fixing that.

**Pull request merged**